src/dispatch/incident_cost_model/service.py

```diff
@@ -0,0 +1,215 @@
+from datetime import datetime
+import logging
+
+from .models import (
+    IncidentCostModel,
+    IncidentCostModelCreate,
+    IncidentCostModelRead,
+    IncidentCostModelUpdate,
+    IncidentCostModelActivity,
+    IncidentCostModelActivityCreate,
+    IncidentCostModelActivityUpdate,
+)
+from dispatch.incident_cost_model import service as incident_cost_model_service
+from dispatch.plugin import service as plugin_service
+from dispatch.project import service as project_service
+
+log = logging.getLogger(__name__)
+
+
+def has_unique_plugin_event(incident_cost_model_in: IncidentCostModelRead):
+    seen = set()
+    for activity in incident_cost_model_in.activities:
+        if activity.plugin_event.id in seen:
+            log.warning(
+                f"Duplicate plugin event id detected. Please ensure all plugin events are unique for each cost model. Duplicate id: {activity.plugin_event.id}"
+            )
+            return False
+        seen.add(activity.plugin_event.id)
+    return True
+
+
+def get_incident_cost_model_activity_by_id(
+    *, db_session, incident_cost_model_activity_id: int
+) -> IncidentCostModelActivity:
+    """Returns an incident cost model based on the given incident cost model id."""
+    return (
+        db_session.query(IncidentCostModelActivity)
+        .filter(IncidentCostModelActivity.id == incident_cost_model_activity_id)
+        .one()
+    )
+
+
+def delete_incident_cost_model_activity(*, db_session, incident_cost_model_activity_id: int):
+    """Deletes an incident cost model activity."""
+    incident_cost_model_activity = get_incident_cost_model_activity_by_id(
+        db_session=db_session, incident_cost_model_activity_id=incident_cost_model_activity_id
+    )
+    db_session.delete(incident_cost_model_activity)
+    db_session.commit()
+
+
+def update_incident_cost_model_activity(
+    *, db_session, incident_cost_model_activity_in: IncidentCostModelActivityUpdate
+):
+    """Updates an incident cost model activity."""
+    incident_cost_model_activity = IncidentCostModelActivity(
+        response_time_seconds=incident_cost_model_activity_in.response_time_seconds,
+        enabled=incident_cost_model_activity_in.enabled,
+        plugin_event_id=incident_cost_model_activity_in.plugin_event.id,
+    )
+
+    incident_cost_model_activity.response_time_seconds = (
+        incident_cost_model_activity_in.response_time_seconds
+    )
+    incident_cost_model_activity.enabled = incident_cost_model_activity_in.enabled
+    incident_cost_model_activity.plugin_event_id = incident_cost_model_activity_in.plugin_event.id
```

Comment:
Shouldn't you fetch the activity using `incident_cost_model_activity_id`, then update it with the values in `incident_cost_model_activity_in`?